plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxConfig.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.influxdb;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.units.Duration;
+import io.airlift.units.MinDuration;
+
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import java.util.Optional;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class InfluxConfig
+{
+    private String endpoint;
+    private String username;
+    private String password;
+    private Duration connectTimeOut = new Duration(10, SECONDS);
+    private Duration readTimeOut = new Duration(60, SECONDS);
+
+    @NotEmpty
+    @Pattern(message = "Invalid endpoint. Expected http:// or https://", regexp = "^https?://.*")
```

Comment:
In case of `https` how would we pass the keystore or truststore configuration ?

Reply:
The connector does not support to https endpoint yet, I will update the comment. 
The official client doesn't have built-in support for SSL:
https://github.com/influxdata/influxdb-java/blob/828298022b1a53c45ce84ccb2e03cb983469c38b/FAQ.md#how-to-use-ssl-client-certificate-authentication